src/zap.ts

```diff
@@ -0,0 +1,55 @@
+import { ChainId, isEthereum, isFantom } from "./chain";
+import { EthAddress, ZeroAddress } from "./helpers";
+import { Token, VaultMetadataOverrides } from "./types";
+
+export type ZapInWith = keyof Token["supported"];
+
+type ZapInDetails = { isZapInSupported: boolean; zapInWith?: ZapInWith };
+
+type ZapInArgs = {
+  chainId: ChainId;
+  token?: Partial<Token>;
+  vaultMetadata: VaultMetadataOverrides | null;
+};
+
+export function getZapInDetails({ chainId, token, vaultMetadata }: ZapInArgs): ZapInDetails {
+  if (!token?.supported || !vaultMetadata) {
```

Comment:
```suggestion
  if (!token?.supported || !vaultMetadata?.zapInWith) {
```
If vault does not have set `zapInWith`, we can just return too

Reply:
Good idea, done with https://github.com/yearn/yearn-sdk/pull/280/commits/d55bcfeed465497b8826f6848f26f63bc77a691c

Reply:
Actually this might not be good because the vaultMetadata when depositing ETH doesn't appear to have a `zapInWith`